protocol/v2/types/ssvshare.go

```diff
@@ -42,14 +56,39 @@ func (s *SSVShare) BelongsToOperator(operatorID spectypes.OperatorID) bool {
 	})
 }
 
-// HasBeaconMetadata checks whether the BeaconMetadata field is not nil.
+// HasBeaconMetadata checks whether SSVShare has been enriched with respective Beacon metadata.
 func (s *SSVShare) HasBeaconMetadata() bool {
-	return s != nil && s.BeaconMetadata != nil
+	return s != nil && s.Status != eth2apiv1.ValidatorStateUnknown
 }
```

Comment:
Now that we've gotten rid of `s.BeaconMetadata` I believe `s.Status != eth2apiv1.ValidatorStateUnknown` check here is sufficient to determine that we've got/fetched metadata from beacon node.